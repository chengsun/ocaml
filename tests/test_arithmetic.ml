let () =
  print_int ((+8) lsr 2); print_newline ();
  print_int ((-8) lsr 2); print_newline ();
  print_int ((-8) lsr 48); print_newline ();
  print_int ((+8) asr 2); print_newline ();
  print_int ((-8) asr 2); print_newline ();

  print_int (44 + (-2)); print_newline ();
  print_int (40 - (-2)); print_newline ();
  print_int (21 * 2); print_newline ();
  print_int ((-84) / (-2)); print_newline ();

  print_float (44. +. (-2.)); print_newline ();
  print_float (40. -. (-2.)); print_newline ();
  print_float (21. *. 2.); print_newline ();
  print_float ((-84.) /. (-2.)); print_newline ();
