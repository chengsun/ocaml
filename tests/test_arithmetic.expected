2
281474976710654
3
2
-2
